package NYT.NChunkServer.NProto;

import "yt/core/misc/proto/guid.proto";

////////////////////////////////////////////////////////////////////////////////

message TReqUpdateChunkRequisition
{
    message TUpdate
    {
        required NYT.NProto.TGuid chunk_id = 1;
        required uint32 chunk_requisition_index = 2;
    }
    message TChunkRequisition
    {
        message TAccountingEntry
        {
            required NYT.NProto.TGuid account_id = 4;
            required int32 medium_index = 1;
            required int32 replication_factor = 2;
            required bool data_parts_only = 3;
            required bool committed = 5;
        }
        required bool vital = 3;
        repeated TAccountingEntry entries = 4;
    }
    message TChunkRequisitionDictItem
    {
        required uint32 index = 1;
        required TChunkRequisition requisition = 2;
    }

    required int32 cell_tag = 1;
    repeated TUpdate updates = 3;
    repeated TChunkRequisitionDictItem chunk_requisition_dict = 4;
}

message TReqConfirmChunkListsRequisitionTraverseFinished
{
    repeated NYT.NProto.TGuid chunk_list_ids = 1;
}

////////////////////////////////////////////////////////////////////////////////
